sdk.yaml

```diff
@@ -14,148 +19,84 @@ Agent Identifier:
-    .synopsis: |
-      Where that string is treated as an API key if it includes a ':' (colon) delimiter,
-      otherwise it is treated as a token string.
-Options:
-  .synopsis: |
-    `ClientOptions` that apply to both client types (REST and Realtime).
-  Agents:
-    .specification: RSC7d6
+    Mechanisms authenticating with the Ably service.
+    Encapsulated by the `AuthOptions` class.
+    Implies the existence of the `clientId` client option, the identifier of the client represented by the client instance.
+  Basic:
+    .documentation:
+      - https://ably.com/docs/core-features/authentication#basic-authentication
+    .specification: [RSA11, RSC1, TO3j1, AO2a]
     .synopsis: |
-      `agents`:
-      For use only by other Ably-authored SDKs, on a need-to-have basis.
-  Authentication:
+      Capability to use a 'raw', full Ably key string, as obtained from the dashboard, for authentication.
+      Specified using the `key` client option and features a key name and key secret.
+      Should only be used server-side in production applications.
+      Client-side applications should always use token authentication in production.
+  Token:
+    .documentation:
+      - https://ably.com/docs/core-features/authentication#token-authentication
+    .specification: [RSA4, RSA14, TO3j4, TO3j11, RSA7a4]
     .synopsis: |
-      `AuthOptions`
+      Capability to have a trusted device, such as one of your own servers possessing an API key configured via the dashboard, create short-lived access tokens for this client.
+      The trusted device uses the API key to distribute time-limited tokens with limited sets of access rights or capabilities, or with specific identities to untrusted clients.
+      The `useTokenAuth` client option can be used to specify that the client library should not use basic authentication,
+      guaranteeing failure if none of `authCallback`, `authUrl`, `token` or `tokenDetails` are also supplied in client options.
+      The `defaultTokenParams` client option overrides client library defaults for `TokenParams`.
     Callback:
       .specification: [RSA4a, RSA4, TO3j5, AO2b]
       .synopsis: |
-        `authCallback`:
-        Service `Auth#requestToken` calls by invoking a callback (`authCallback`).
-    Key:
-      .specification: [RSA11, TO3j1, AO2a]
-      .synopsis: |
-        `key`:
-        Authenticate using a full Ably key string, as obtained from the dashboard,
-        featuring a key name and key secret.
-    Query Time:
-      .specification: [TO3j10, RSA9d, RSA10k, AO2g]
-      .synopsis: |
-        `queryTime`:
-        Query the Ably system for the current time when making token requests.
-    Token Details:
-      .specification: [RSA4a, RSA4, TO3j3]
-      .synopsis: |
-        `tokenDetails`
-    Token Request:
-      .specification: [TO3j2]
+        Capability for the application to service `Auth#requestToken` calls by supplying a callback using the `authCallback` client option.
```

Comment:
Rather than "Capability for the application to do X", I think we should just say "X" (since this whole document is about capabilities to do X), so here:

"Supply a callback as the `authCallback` client option to obtain authentication tokens using user-supplied logic."

Reply:
I take your point, however I notice that I use the word "capability" in a lot of other places in this document. I will, therefore, address this under a new pull request - outside of the scope of this pull request. I've created https://github.com/ably/features/issues/44 to track that.

Reply:
Resolving this conversation on @lmars' behalf, as I've deferred addressing the comment until later and wish to close this pull request cleanly.